pkg/stdlib/objects/merge_recursive.go

```diff
@@ -0,0 +1,66 @@
+package objects
+
+import (
+	"context"
+
+	"github.com/MontFerret/ferret/pkg/runtime/core"
+	"github.com/MontFerret/ferret/pkg/runtime/values"
+)
+
+// MergeRecursive recursively merge the given objects into a single object.
+// @params objs (Objects) - objects to merge.
+// @returns (Object) - Object created by merging.
+func MergeRecursive(_ context.Context, args ...core.Value) (core.Value, error) {
+	err := core.ValidateArgs(args, 1, core.MaxArgs)
+	if err != nil {
+		return values.None, err
+	}
+
+	for _, arg := range args {
+		if err = core.ValidateType(arg, core.ObjectType); err != nil {
+			return values.None, err
+		}
+	}
+
+	merged := values.NewObject()
+
+	for _, arg := range args {
+		merged = merge(merged, arg.(*values.Object)).(*values.Object)
+	}
+
+	return merged.Clone(), nil
+}
+
+func merge(src, dst core.Value) core.Value {
```

Comment:
If it's a private method and you use it for merging objects, why don't you define its arguments and the output as the``*values.Object`` type?

Reply:
because this function can take values of different types at runtime